### PR TITLE
Add Sorting to Contest Page 2

### PR DESCRIPTION
**Pull request by ComputerGenius152**
Sorry for the [duplicate pull request](https://github.com/DMOJ/online-judge/pull/1474), but while squashing my commits on the previous one I force-pushed to master and Github automatically closed my pull request. Once again, this closes Issue #804.

### CODE REVIEW

File: judge/views/contests.py [hunk 1]
```diff
@@ -97,13 +100,17 @@ def get_context_data(self, **kwargs):
...
-        active.sort(key=attrgetter('end_time'))
+        active.sort(key=attrgetter('end_time', 'key'))
+        present.sort(key=attrgetter('end_time', 'key'))
         future.sort(key=attrgetter('start_time'))
         context['active_participations'] = active
         context['current_contests'] = present
         context['future_contests'] = future
         context['now'] = self._now
         context['first_page_href'] = '.'
+        context['page_suffix'] = '#past-contests'
```
Comment (Ninjaclasher) [inline judge/views/contests.py:111]: This shouldn't be here.

File: templates/contest/list.html [hunk 2]
```diff
@@ -257,7 +257,7 @@ <h4>{{ _('Upcoming Contests') }}</h4>
         <br>
 
         {% if past_contests %}
-            <h4>{{ _('Past Contests') }}</h4>
+            <h4 id="past-contests">{{ _('Past Contests') }}</h4>
```
Comment (Ninjaclasher) [inline templates/contest/list.html:260]: This shouldn't be here.

Comment (ComputerGenius152) [reply templates/contest/list.html:260]: This is in the code currently: https://github.com/DMOJ/online-judge/blob/master/templates/contest/list.html#L260.

File: judge/views/contests.py (hunk 1 above)
Comment (ComputerGenius152) [reply judge/views/contests.py]: Fixed.

**Pull request merged**